### PR TITLE
change default view to max extent of valid layers

### LINKED ISSUE #301: Change geosafe default view to max extent of valid layers
# Problem
When you enter the GeoSAFE analysis screen, the default view is the whole world, even if the available data sets cover only a small part. 

e.g. the INGC view should go straight to Mozambique

# Solution
Get the total extent of all valid and avaiable GeoSAFE layers and choose a centre and zoom level based on that.

The 'zoom to global view' control shoud actually do the same thing and be renamed 'zoom to full extent', like the standard GIS tool to 'zoom to full extent of all layers' rather than 'the whole world'

### PR DESCRIPTION
**Pull request by boney-bun**
fix #301

### CODE REVIEW

**Comment by boney-bun** [comment]
the PR in actions:
- layers in indonesia and africa:
geosafe 301-1

- only 1 layer:
geosafe 301-2

- layers in africa:
geosafe 301-3

**Comment by gubuntu** [comment]
LGTM

File: templates/geosafe/analysis/dynamic_scripts/create/main.html [hunk 1]
```diff
@@ -356,6 +357,36 @@
...
+                current_bounds.getSouthWest().lng,
+                layer_bounds.getSouthWest().lng)];
+        var new_ne = [
+            Math.max(
+                current_bounds.getNorthEast().lat,
+                layer_bounds.getNorthEast().lat),
+            Math.max(
+                current_bounds.getNorthEast().lng,
+                layer_bounds.getNorthEast().lng)];
+        map.fitBounds([new_sw, new_ne]);
```
Comment (lucernae) [inline templates/geosafe/analysis/dynamic_scripts/create/main.html]: Just directly fitBounds to hazard_exposure_group.getBounds() with padding options.

Call this when hazard_exposure_group contains all the layers (last call).

**Review by lucernae — APPROVED** [review]
LGTM with some comments. Thanks @boney-bun

**Comment by boney-bun** [comment]
@lucernae i've been refactored the code so that it can now require less calculation in map bounds.
since the map.fitbounds is inside an async function, the challenge is the function may not be executed in order. 
Hence, you would expect that sometimes the map bounds will be different.

here are the illustation:

1. some african layers:
geosafe301-3

2. some african layers + a nepal layer:
geosafe301-2

3. some african layers + a nepal layer + padang layer: 
geosafe301

what do you think?

File: templates/geosafe/analysis/create_base.html [hunk 2]
```diff
@@ -680,31 +680,35 @@ <h4 class="modal-title" id="myModalLabel">{% trans "Upgrade Your Browser" %}</h4
...
                     layer_id = aggregation_layer_cbo[0][loop_aggregation].value;
                     // first combo option is an empty tag.
                     if (layer_id != "") {
-                        show_marker(layer_id, "aggregation");
+                        show_marker(layer_id, "aggregation", current_loop);
                     }
                 }
                 //use markers as the default
-                map.addLayer(hazard_exposure_marker_groups);
+                //map.addLayer(hazard_exposure_marker_groups);
```
Comment (lucernae) [inline templates/geosafe/analysis/create_base.html:710]: Delete if not needed to save indihome bandwidth.

File: templates/geosafe/analysis/dynamic_scripts/create/main.html [hunk 3]
```diff
@@ -365,12 +366,12 @@
      * @param layer_id
      * @param category
      */
-    function show_marker(layer_id, category) {
+    function show_marker(layer_id, category, current_loop) {
```
Comment (lucernae) [inline templates/geosafe/analysis/dynamic_scripts/create/main.html:369]: I'm not sure what current_loop means. Could you add notes to above JSDocs?

**Review by lucernae — APPROVED** [review]
I don't understand most of the logic @boney-bun :D
But you can merge it if you already tested it.

**Comment by gubuntu** [comment]
@boney-bun please merge

**Comment by boney-bun** [comment]
i'm going to merge this.
feel free to open this ticket again in the future.

**Pull request merged**